src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -198,22 +232,88 @@ private static OpenApiSchema CreateCollectionSchema(ODataContext context, OpenAp
                     }
                 }
             };
-            if (context.Settings.EnablePagination)
-            {
-                properties.Add(
-                    "@odata.nextLink",
-                    new OpenApiSchema
-                    {
-                        Type = "string"
-                    });
-            }
 
-            return new OpenApiSchema
+            OpenApiSchema baseSchema = new()
             {
-                Title = $"Collection of {typeName}",
-                Type = "object",
+                Type = Constants.ObjectType,
                 Properties = properties
             };
+
+            // @odata.nextLink
+            OpenApiSchema paginationSchema = new()
+            {
+                UnresolvedReference = true,
+                Reference = new OpenApiReference
+                {
+                    Type = ReferenceType.Schema,
+                    Id = Constants.BaseCollectionPaginationResponse
+                }
+            };
+
+            // @odata.count
+            OpenApiSchema countSchema = new()
+            {
+                UnresolvedReference = true,
+                Reference = new OpenApiReference
+                {
+                    Type = ReferenceType.Schema,
+                    Id = Constants.BaseCollectionCountResponse
+                }
+            };
+
+            // @odata.nextLink + @odata.count
+            OpenApiSchema paginationCountSchema = new()
+            {
+                UnresolvedReference = true,
+                Reference = new OpenApiReference
+                {
+                    Type = ReferenceType.Schema,
+                    Id = Constants.BaseCollectionPaginationCountResponse
+                }
+            };
+
+            OpenApiSchema colSchema;
+            if (context.Settings.EnablePagination && context.Settings.EnableCount)
```

Comment:
```suggestion
            if (context.Settings.EnablePagination || context.Settings.EnableCount)
```

Reply:
What about when both settings are set to `true`?

Reply:
the base type will have both properties in that case

Reply:
If in either case, both properties are added, then why have the 2 different settings? Why not just have 1?

Reply:
because as you mentioned earlier, you can want count to be part of your conversion, paging, or both. They are not overlapping or mutually exclusive.

Reply:
This was borne from your earlier comment :)
https://github.com/microsoft/OpenAPI.NET.OData/pull/270#discussion_r959517460

Reply:
yep, thanks for double checking. I do contradict myself sometimes. But I think we have a flexible and consistent solution which is low maintenance right now. Unless you disagree, we're good to merge IMHO.

Reply:
I'm good with it.